src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -323,43 +330,49 @@ public List<JobModel> enumerateJobs(
+      logger.info("Getting flight batch {} with nextPageToken {}", numTimesQueried, nextPageToken);
+      FlightEnumeration filterResults =
+          stairway.getFlights(nextPageToken, FLIGHT_SEARCH_BATCH_SIZE, filter);
+      if (filterResults.getFlightStateList().size() == 0) {
         break;
       }
-      filterResults.forEach(
-          flightState -> {
-            if (userLaunchedFlight(flightState, userReq)) {
+
+      for (var flightState : filterResults.getFlightStateList()) {
+        if (userLaunchedFlight(flightState, userReq)) {
+          flightStateList.add(flightState);
+        } else {
+          FlightMap inputParameters = flightState.getInputParameters();
+          IamResourceType resourceType =
+              inputParameters.get(JobMapKeys.IAM_RESOURCE_TYPE.getKeyName(), IamResourceType.class);
+          String resourceId =
+              inputParameters.get(JobMapKeys.IAM_RESOURCE_ID.getKeyName(), String.class);
+          IamAction action =
+              inputParameters.get(JobMapKeys.IAM_ACTION.getKeyName(), IamAction.class);
+          if (resourceType != null & resourceId != null && action != null) {
+            String key = String.format("%s-%s", resourceType, resourceId);
+            List<String> userRoles = roleMap.get(key);
+            if (userRoles == null) {
+              userRoles = samService.listActions(userReq, resourceType, resourceId);
+              roleMap.put(key, userRoles);
+            }
+            if (userRoles.contains(action.toString())) {
               flightStateList.add(flightState);
-            } else {
-              FlightMap inputParameters = flightState.getInputParameters();
-              IamResourceType resourceType =
-                  inputParameters.get(
-                      JobMapKeys.IAM_RESOURCE_TYPE.getKeyName(), IamResourceType.class);
-              String resourceId =
-                  inputParameters.get(JobMapKeys.IAM_RESOURCE_ID.getKeyName(), String.class);
-              IamAction action =
-                  inputParameters.get(JobMapKeys.IAM_ACTION.getKeyName(), IamAction.class);
-              if (resourceType != null & resourceId != null && action != null) {
-                String key = String.format("%s-%s", resourceType, resourceId);
-                List<String> userRoles = roleMap.get(key);
-                if (userRoles == null) {
-                  userRoles = samService.listActions(userReq, resourceType, resourceId);
-                  roleMap.put(key, userRoles);
-                }
-                if (userRoles.contains(action.toString())) {
-                  flightStateList.add(flightState);
-                }
-              }
             }
-          });
-      start = end;
-      end += offset + limit;
+          }
+        }
+        // Stop processing as soon as we've got enough to return
+        if (flightStateList.size() == offset + limit) {
```

Comment:
is `==` correct here or do you want `>=`? The outer loop is using `flightStateList.size() < offset + limit`, not `!=`.

Reply:
Yeah, I mean, really shouldn't skip the case where it's `==`